src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java

```diff
@@ -150,17 +150,32 @@ public static String fillAuthentication(String settingsContent, Map<String, Stan
     /**
      * Removes all childs
      * 
-     * @param serverNode
+     * @param serversNode
      *            the node to remove all childs from
      */
-    private static void removeAllChilds(final Node serverNode) {
-        final NodeList childNodes = serverNode.getChildNodes();
-        for (int i = 0; i < childNodes.getLength(); i++) {
-            final Node child = childNodes.item(i);
-            serverNode.removeChild(child);
-            --i;
+    private static void removeAllChilds(final Node serversNode, final Set<String> credentialKeys, final Boolean replaceAll) {
+        final NodeList serverNodes = serversNode.getChildNodes();
+        for (int i = 0; i < serverNodes.getLength(); i++) {
+            final Node server = serverNodes.item(i);
+            String serverId = getServerId(server);
+            if (Boolean.TRUE.equals(replaceAll) || (credentialKeys.contains(serverId) && Boolean.FALSE.equals(replaceAll))) {
```

Comment:
Good point, thanks for that simplification.


From: Dominik Bartholdi [mailto:notifications@github.com]
Sent: Tuesday, 12 May 2015 2:15 AM
To: jenkinsci/config-file-provider-plugin
Cc: de Klepper, Nathan
Subject: Re: [config-file-provider-plugin] add option to preserve servers in server.xml file not found in the provi... (#10)


In src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java<https://github.com/jenkinsci/config-file-provider-plugin/pull/10#discussion_r30054117>:

>       *            the node to remove all childs from

>       */

> -    private static void removeAllChilds(final Node serverNode) {

> -        final NodeList childNodes = serverNode.getChildNodes();

> -        for (int i = 0; i < childNodes.getLength(); i++) {

> -            final Node child = childNodes.item(i);

> -            serverNode.removeChild(child);

> -            --i;

> +    private static void removeAllChilds(final Node serversNode, final Set<String> credentialKeys, final Boolean replaceAll) {

> +        final NodeList serverNodes = serversNode.getChildNodes();

> +        for (int i = 0; i < serverNodes.getLength(); i++) {

> +            final Node server = serverNodes.item(i);

> +            String serverId = getServerId(server);

> +            if (Boolean.TRUE.equals(replaceAll) || (credentialKeys.contains(serverId) && Boolean.FALSE.equals(replaceAll))) {

is && Boolean.FALSE.equals(replaceAll) really required? if serverId is containedin the key, then we remove it anyway

—
Reply to this email directly or view it on GitHub<https://github.com/jenkinsci/config-file-provider-plugin/pull/10/files#r30054117>.
Australia Post is committed to providing our customers with excellent service. If we can assist you in any way please telephone 13 13 18 or visit our website.

The information contained in this email communication may be proprietary, confidential or legally professionally privileged. It is intended exclusively for the individual or entity to which it is addressed. You should only read, disclose, re-transmit, copy, distribute, act in reliance on or commercialise the information if you are authorised to do so. Australia Post does not represent, warrant or guarantee that the integrity of this email communication has been maintained nor that the communication is free of errors, virus or interference.

If you are not the addressee or intended recipient please notify us by replying direct to the sender and then destroy any electronic or paper copy of this message. Any views expressed in this email communication are taken to be those of the individual sender, except where the sender specifically attributes those views to Australia Post and is authorised to do so.

Please consider the environment before printing this email.